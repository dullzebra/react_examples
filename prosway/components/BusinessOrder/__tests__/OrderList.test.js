import React from 'react';
import renderer from 'react-test-renderer';

import {OrdersList} from '../OrdersList/OrdersList';
jest.mock('../OrdersList/OrderItem', () => ({item}) => `OrderItem id: ${item.id} , status: ${item.status}`);
jest.mock('react-modal', () => 'Modal');

const expectedProps = {
  items: [],
  weekdays: [],
};

const expectedProps2 = {
  items: [{
    id: 'abc',
    status: '+',
  },
  {
    id: 'abc123',
    status: '?',
  },
  {
    id: 'abc000',
    status: '-',
  }],
  weekdays: [],
};

const setupComponent = (props) => {
  return renderer.create(
    <OrdersList {...props} />
  );  
};

describe('<OrdersList />', () => {
  it('should render w/o crashing', () => {
    const component = setupComponent(expectedProps);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render list of items with statuses (?|+)', () => {
    const component = setupComponent(expectedProps2);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should open <OrderUserContact> with data on "openPopup" method call', () => {
    const component = setupComponent(expectedProps2);
    component.getInstance().openPopup('Иван Иваныч', '+7 919 9111111', 'ivanych@procvet.io');
    const modal = component.root.find(el => el.type === 'Modal');
    const tree = component.toJSON();

    expect(modal.props.isOpen).toBe(true);
    expect(tree).toMatchSnapshot();
  });
});
