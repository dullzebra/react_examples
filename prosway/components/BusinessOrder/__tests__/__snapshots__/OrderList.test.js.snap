// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrdersList /> should open <OrderUserContact> with data on "openPopup" method call 1`] = `
Array [
  <ul
    className="OrdersList"
  >
    <li
      className="OrdersListItem"
    >
      <div
        className="OrdersListItemIndex"
      >
        1
        .
      </div>
      OrderItem id: abc , status: +
    </li>
    <li
      className="OrdersListItem"
    >
      <div
        className="OrdersListItemIndex"
      >
        2
        .
      </div>
      OrderItem id: abc123 , status: ?
    </li>
  </ul>,
  <Modal
    bodyOpenClassName="ReactModal__Body--open"
    className="OrdersListModal"
    isOpen={true}
    onRequestClose={[Function]}
    overlayClassName="OrdersListModalOverlay"
  >
    <div
      className="OrderUserContact"
    >
      <div
        className="Title"
      >
        Иван Иваныч
      </div>
      <p
        className="Contact"
      >
        Телефон: 
        <span
          className="ContactValue"
        >
          +7 919 9111111
        </span>
      </p>
      <p
        className="Contact"
      >
        Почта: 
        <a
          href="mailto:ivanych@procvet.io"
        >
          ivanych@procvet.io
        </a>
      </p>
      <div
        className="Line"
      />
      <div
        className="Buttons"
      >
        <button
          className="ButtonTransparent"
          disabled={undefined}
          onClick={[Function]}
          type="button"
        >
          закрыть
        </button>
      </div>
    </div>
  </Modal>,
]
`;

exports[`<OrdersList /> should render list of items with statuses (?|+) 1`] = `
Array [
  <ul
    className="OrdersList"
  >
    <li
      className="OrdersListItem"
    >
      <div
        className="OrdersListItemIndex"
      >
        1
        .
      </div>
      OrderItem id: abc , status: +
    </li>
    <li
      className="OrdersListItem"
    >
      <div
        className="OrdersListItemIndex"
      >
        2
        .
      </div>
      OrderItem id: abc123 , status: ?
    </li>
  </ul>,
  <Modal
    bodyOpenClassName="ReactModal__Body--open"
    className="OrdersListModal"
    isOpen={false}
    onRequestClose={[Function]}
    overlayClassName="OrdersListModalOverlay"
  >
    <div
      className="OrderUserContact"
    >
      <div
        className="Title"
      />
      <p
        className="Contact"
      >
        Телефон: 
        <span
          className="ContactValue"
        />
      </p>
      <p
        className="Contact"
      >
        Почта: 
        <a
          href="mailto:undefined"
        />
      </p>
      <div
        className="Line"
      />
      <div
        className="Buttons"
      >
        <button
          className="ButtonTransparent"
          disabled={undefined}
          onClick={[Function]}
          type="button"
        >
          закрыть
        </button>
      </div>
    </div>
  </Modal>,
]
`;

exports[`<OrdersList /> should render w/o crashing 1`] = `
Array [
  <ul
    className="OrdersList"
  />,
  <Modal
    bodyOpenClassName="ReactModal__Body--open"
    className="OrdersListModal"
    isOpen={false}
    onRequestClose={[Function]}
    overlayClassName="OrdersListModalOverlay"
  >
    <div
      className="OrderUserContact"
    >
      <div
        className="Title"
      />
      <p
        className="Contact"
      >
        Телефон: 
        <span
          className="ContactValue"
        />
      </p>
      <p
        className="Contact"
      >
        Почта: 
        <a
          href="mailto:undefined"
        />
      </p>
      <div
        className="Line"
      />
      <div
        className="Buttons"
      >
        <button
          className="ButtonTransparent"
          disabled={undefined}
          onClick={[Function]}
          type="button"
        >
          закрыть
        </button>
      </div>
    </div>
  </Modal>,
]
`;
